src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/JpaGoalPersistenceAdapter.kt

```diff
@@ -41,7 +41,8 @@ class JpaGoalPersistenceAdapter(
       actionPlanRepository.saveAndFlush(this)
     }
 
-    return goalMapper.fromEntityToDomain(goalEntity)
+    val persisted = actionPlanEntity.goals!!.first { it.reference == goalEntity.reference }
```

Comment:
Just out of interest was there a problem/bug with the original impl? Or is this change just a code tidying / making it easier to read exercise ?

Reply:
It's to ensure the returned goal has the "JPA" fields populated.